x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java

```diff
@@ -419,19 +431,169 @@ protected Object generateRandomInputValue(MappedFieldType ft) {
 
     @Override
     protected SyntheticSourceSupport syntheticSourceSupport(boolean ignoreMalformed) {
-        throw new AssumptionViolatedException("not supported");
+        return syntheticSourceSupport(ignoreMalformed, false);
+    }
+
+    @Override
+    protected SyntheticSourceSupport syntheticSourceSupport(boolean ignoreMalformed, boolean columnReader) {
+        return new SyntheticSourceSupport() {
+            private final boolean ignoreZValue = usually();
+            private final CartesianPoint nullValue = usually() ? null : randomCartesianPoint();
+
+            @Override
+            public boolean preservesExactSource() {
+                return true;
+            }
+
+            @Override
+            public SyntheticSourceExample example(int maxVals) {
+                if (randomBoolean()) {
+                    Value v = generateValue();
+
+                    if (v.point == null) {
+                        return new SyntheticSourceExample(v.representation(), v.representation(), null, this::mapping);
+                    } else if (columnReader) {
+                        return new SyntheticSourceExample(v.representation(), v.representation(), encode(v.point()), this::mapping);
+                    }
+                    return new SyntheticSourceExample(v.representation(), v.representation(), v.point().toWKT(), this::mapping);
+                }
+                List<Value> values = randomList(1, maxVals, this::generateValue);
+                var representations = values.stream().map(Value::representation).toList();
+
+                if (columnReader) {
+                    // When reading doc-values, the block is a list of encoded longs
+                    List<Long> outBlockList = values.stream()
+                        .map(Value::point)
+                        .filter(Objects::nonNull)
+                        .map(this::encode)
+                        .sorted()
+                        .toList();
+                    Object outBlock = outBlockList.size() == 1 ? outBlockList.get(0) : outBlockList;
+                    return new SyntheticSourceExample(representations, representations, outBlock, this::mapping);
+                } else {
+                    // When reading row-stride, the block is a list of WKT encoded BytesRefs
+                    List<String> outBlockList = values.stream()
+                        .map(Value::point)
+                        .filter(Objects::nonNull)
+                        .map(CartesianPoint::toWKT)
+                        .toList();
+                    Object outBlock = outBlockList.size() == 1 ? outBlockList.get(0) : outBlockList;
```

Comment:
Can the list be empty, e.g. if all values are null?

Reply:
Yes but it is handled in test code separately.